### PR TITLE
Update Pipeline SDK overview page

### PR DESCRIPTION
**Pull request by numerology**
Also removed the 'out of date' banner.

### CODE REVIEW

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 1]
```diff
@@ -117,9 +116,14 @@ The Kubeflow Pipelines SDK includes the following packages:
...
-  * `kfp.Client.pipeline_uploads.upload_pipeline_version` uploads a local file to create a pipeline version. [Follow an example to learn more about creating a pipeline version](/docs/pipelines/tutorials/sdk-examples)
-
-* [`kfp.notebook`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.notebook.html)
+  * `kfp.Client.create_run_from_pipeline_func` compiles a pipeline function and submits it
+    for execution on KFP.
+  * `kfp.Client.create_run_from_pipeline_package` runs a local pipeline package on KFP.
+  * `kfp.Client.upload_pipeline` uploads a local file to create a new pipeline in KFP.
+  * `kfp.Client.upload_pipeline_version` uploads a local file to create a pipeline version. [Follow an example to learn more about creating a pipeline version](/docs/pipelines/tutorials/sdk-examples)
+
+* [`kfp.notebook`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.notebook.html) contains magic to help working with Kubeflow Pipelines in a Python
```
Comment (Ark-kun) [inline content/en/docs/pipelines/sdk/sdk-overview.md:125]: I think this no longer exists.

Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md]: Removed. Resolving

**Comment by joeliedtke** [comment]
Preview: 
https://deploy-preview-2167--competent-brattain-de2d6d.netlify.app/docs/pipelines/sdk/sdk-overview/

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 2]
```diff
@@ -107,6 +94,18 @@ The Kubeflow Pipelines SDK includes the following packages:
...
+  * [`kfp.dsl.ParallelFor`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ParallelFor)
+    represents a parallel for loop over a static or dynamic set of items in a pipeline.
+    Each iteration of the for loop will be executed in parallel.
+  
+  * [`kfp.dsl.ExitHandler`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ExitHandler)
+    represents an exit handler that is invoked upon exiting a group of ops. A typical
+    usage of `ExitHandler` is garbage collection. Currently, one pipeline can
+    only include at most one `ExitHandler`.
+  
+  * [`kfp.dsl.Condition`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ResourceOp)
```
Comment (joeliedtke) [inline content/en/docs/pipelines/sdk/sdk-overview.md:106]: Update link:

https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.Condition

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 3]
```diff
@@ -117,9 +116,11 @@ The Kubeflow Pipelines SDK includes the following packages:
     [experiment](/docs/pipelines/concepts/experiment/) and returns an
     experiment object.
   * `kfp.Client.run_pipeline` runs a pipeline and returns a run object.
-  * `kfp.Client.pipeline_uploads.upload_pipeline_version` uploads a local file to create a pipeline version. [Follow an example to learn more about creating a pipeline version](/docs/pipelines/tutorials/sdk-examples)
-
-* [`kfp.notebook`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.notebook.html)
+  * `kfp.Client.create_run_from_pipeline_func` compiles a pipeline function and submits it
+    for execution on KFP.
```
Comment (joeliedtke) [inline content/en/docs/pipelines/sdk/sdk-overview.md:120]: Do not abbreviate Kubeflow Pipelines to KFP. Please fix on this line and the next two lines.

KFP -> Kubeflow Pipelines

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 4]
```diff
@@ -107,6 +94,18 @@ The Kubeflow Pipelines SDK includes the following packages:
...
+    Each iteration of the for loop will be executed in parallel.
+  
+  * [`kfp.dsl.ExitHandler`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ExitHandler)
+    represents an exit handler that is invoked upon exiting a group of ops. A typical
+    usage of `ExitHandler` is garbage collection. Currently, one pipeline can
+    only include at most one `ExitHandler`.
+  
+  * [`kfp.dsl.Condition`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ResourceOp)
+    represents a group of ops, that will only be executed when a certain condition is met.
+    The condition specified can be either determined at compile time or at runtime.
```
Comment (joeliedtke) [inline content/en/docs/pipelines/sdk/sdk-overview.md:108]: If the condition is determined at compile time, is the condition still evaluated as part of the DAG? Or is it left out of the graph?

**Review by joeliedtke — CHANGES_REQUESTED** [review]
Thanks for your patience! I've added a few suggestions and issues that need to be updated. Otherwise this looks great.

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 5]
```diff
@@ -24,26 +20,15 @@ workflow and how the components interact with each other.
 The Kubeflow Pipelines SDK includes the following packages:
 
 * [`kfp.compiler`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.compiler.html)
-  includes classes and methods for building Docker container images for your
-  pipeline components. Methods in this package include, but are not limited
+  includes classes and methods for compiling pipeline Python DSL into workflow yaml spec
```
[the remark this answers is not in the document]
Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md:23]: Done.

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 6]
```diff
@@ -76,7 +61,9 @@ The Kubeflow Pipelines SDK includes the following packages:
   the following:
 
   * `kfp.dsl.ContainerOp` represents a pipeline task (op) implemented by a
```
[the remark this answers is not in the document]
Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md:63]: Done

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 7]
```diff
@@ -107,6 +94,18 @@ The Kubeflow Pipelines SDK includes the following packages:
     represents a volume used to pass data between pipeline steps. `ContainerOp`s 
     can mount a `PipelineVolume` either via the constructor's argument 
     `pvolumes` or `add_pvolumes()` method.
+  * [`kfp.dsl.ParallelFor`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ParallelFor)
+    represents a parallel for loop over a static or dynamic set of items in a pipeline.
+    Each iteration of the for loop will be executed in parallel.
```
[the remark this answers is not in the document]
Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md:99]: Thanks! Done

File: content/en/docs/pipelines/sdk/sdk-overview.md [hunk 8]
```diff
@@ -107,6 +94,18 @@ The Kubeflow Pipelines SDK includes the following packages:
...
+  
+  * [`kfp.dsl.ExitHandler`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ExitHandler)
```
[the remark this answers is not in the document]
Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md:101]: Done. Removed.

File: content/en/docs/pipelines/sdk/sdk-overview.md (hunk 2 above)
Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md:106]: Good catch thanks!

File: content/en/docs/pipelines/sdk/sdk-overview.md (hunk 4 above)
Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md:108]: Sorry I double checked it turns out that only boolean expression determined at runtime can be used here. Updated.

File: content/en/docs/pipelines/sdk/sdk-overview.md (hunk 3 above)
Comment (numerology) [reply content/en/docs/pipelines/sdk/sdk-overview.md:120]: Done. Also applied to other places.

**Review by numerology — COMMENTED** [review]
Thanks! PTAL @joeliedtke

**Pull request merged**